Data/ByteString/Builder/RealFloat/F2S.hs

```diff
@@ -20,12 +21,16 @@ import Data.ByteString.Builder.Internal (Builder)
 import Data.ByteString.Builder.Prim (primBounded)
 import Data.ByteString.Builder.RealFloat.Internal
 import GHC.Int (Int32(..))
-import GHC.Ptr (Ptr(..))
 import GHC.Word (Word32(..), Word64(..))
 
+#if !PURE_HASKELL
```

Comment:
There is probably too much CPP to consider, and this suggestion would be better left for another PR. But when working on klister, I hid all the CPP behind a module to cleanly separate it as an implementation detail from the actual code that uses it. Here is an example: [this](https://github.com/gelisam/klister/blob/67d4eedf8a2c7b4ca67ac01e8813052c29cd0557/src/Scope.hs#L4) module exports a data type that is changed by CPP, so that other modules can use the type without knowing about the CPP.

Reply:
I also think it would be better left for another PR dedicated to it if needed.